assets/app/scripts/controllers/util/oauth.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:OAuthController
+ * @description
+ * # OAuthController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('OAuthController', function ($location, $q, RedirectLoginService, DataService, AuthService) {
+
+    var debug = true;
+
+    RedirectLoginService.finish()
+    .then(function(token, then) {
+      // Redirect to the root
+      if (!token) {
+        if (debug) { console.debug("OAuthController, no token or error, redirecting to /"); }
+        $location.url('/');
```

Comment:
currently won't this loop since "/" is the project listing which requests user?  if so we should probably add a comment / TODO here